src/searchers/mod.rs

```diff
@@ -82,7 +84,9 @@ mod tests {
     #[test]
     fn perform_decoding_succeeds_empty_string() {
         // Some decoders like base64 return even when the string is empty.
-        let result = perform_decoding("");
+        let mut dc = DecoderResult::default();
+        dc.text = vec!["".to_string()];
```

Comment:
can we create a `new` ( or `with_text()` ) methodfor `DecoderResult` which can accept &str / String :

```rust
let dc = DecoderResult::new("");
```